<!--
/******************************************************************************/
/*                                                                            */
/*  Copyright (C) 2018, FLIR Systems                                          */
/*  All rights reserved.                                                      */
/*                                                                            */
/******************************************************************************/
-->
<?xml version="1.0"?>
<!-- Test launcher to see the discrepancy between left and right 
actual frame capture times -->
<launch>
  <arg name="namespace" default="flir_adk"/>
  <arg name="left_frame_id" default="bf1"/>
  <arg name="right_frame_id" default="bf2"/>

  <arg name="left_ip" default="169.254.87.157"/>
  <arg name="right_ip" default="169.254.82.142"/>

  <arg name="left_namespace" default="left" />
  <arg name="right_namespace" default="right" />

  <arg name="frame_rate" default="60.0"/>

  <arg name="camera_info_url" default="package://flir_adk_ethernet/example_calibrations/BlackFlyS.yaml"/>

  <group ns="$(arg namespace)">
    <node pkg="nodelet" 
          type="nodelet"
          name="nodelet_manager"
          args="manager"
          output="screen"/>


    <node pkg="nodelet" 
          type="nodelet"
          name="left_camera_node"
          args="load flir_adk_ethernet/SyncCameraController nodelet_manager"
          output="screen">
      <remap from="image_raw" to="$(arg left_namespace)/image_raw" />
      <remap from="camera_info_url" to="$(arg left_namespace)/camera_info_url" />
      <remap from="set_camera_info" to="$(arg left_namespace)/set_camera_info" />
      <remap from="actual_timestamp" to="$(arg left_namespace)/actual_timestamp" />
      
      <param name="frame_id" type="str" value="$(arg left_frame_id)"/>
      <param name="ip_addr" type="str" value="$(arg left_ip)"/>
      <param name="camera_info_url" type="str" value="$(arg camera_info_url)"/>
    </node>

    <node pkg="nodelet" 
          type="nodelet"
          name="right_camera_node"
          args="load flir_adk_ethernet/SyncCameraController nodelet_manager"
          output="screen">
      <remap from="image_raw" to="$(arg right_namespace)/image_raw" />
      <remap from="camera_info_url" to="$(arg right_namespace)/camera_info_url" />
      <remap from="set_camera_info" to="$(arg right_namespace)/set_camera_info" />
      <remap from="actual_timestamp" to="$(arg right_namespace)/actual_timestamp" />

      <param name="frame_id" type="str" value="$(arg right_frame_id)"/>
      <param name="ip_addr" type="str" value="$(arg right_ip)"/>
      <param name="camera_info_url" type="str" value="$(arg camera_info_url)"/>
    </node>

    <node pkg="nodelet"
          type="nodelet"
          name="time_sync"
          args="load flir_adk_ethernet/TimeSynchronizer nodelet_manager">
      <param name="frame_rate" type="double" value="$(arg frame_rate)"/>
    </node>

    <node pkg="nodelet"
          type="nodelet"
          name="time_diff"
          args="load flir_adk_ethernet/TimeDifference nodelet_manager">
    </node>
  </group>

  <node name="left_view"
        pkg="image_view"
        type="image_view"
        output="screen">
    <remap from="image" to="flir_adk/left/image_raw" />
  </node>
  <node name="right_view"
        pkg="image_view"
        type="image_view"
        output="screen">
    <remap from="image" to="flir_adk/right/image_raw" />
  </node>
</launch>
